kuma/core/decorators.py

```diff
@@ -136,3 +137,17 @@ def is_superuser(u):
 
 superuser_required = user_passes_test(is_superuser)
 #: A decorator to use for requiring a superuser
+
+
+def block_user_agents(view_func):
+    blockable_user_agents = []
+    if hasattr(settings, 'BLOCKABLE_USER_AGENTS'):
+        blockable_user_agents = settings.BLOCKABLE_USER_AGENTS
+
+    def agent_blocked_view(request, *args, **kwargs):
+        if 'HTTP_USER_AGENT' in request.META:
```

Comment:
Don't do a containment check and then get the data again two lines later in every loop over the patterns, this is a high traffic decorator, stuff like that will safe cycles. Use instead:

```
HTTP_USER_AGENT = request.META.get('HTTP_USER_AGENT')
for user_agent_pattern in blockable_user_agents:
    if user_agent_regex.match(HTTP_USER_AGENT):
        raise PermissionDenied
# ...
```

The use of match instead of search should make this faster as well.